csharp/src/Ice/OutgoingAsync.cs

```diff
@@ -593,17 +595,9 @@ protected override bool ResponseImpl(bool userThread, bool ok, bool invoke)
             return base.ResponseImpl(userThread, ok, invoke);
         }
 
+        // TODO: add facet and operation to message
         public void RunTimerTask()
-        {
-            if (Proxy.IceReference.GetInvocationTimeout() == -2)
-            {
-                Cancel(new Ice.ConnectionTimeoutException());
-            }
-            else
-            {
-                Cancel(new Ice.InvocationTimeoutException());
-            }
-        }
+            => Cancel(new TimeoutException($"invocation on remote Ice object `{Proxy.Identity}' timed out"));
```

Comment:
Using the same exception effectively removes the support for `invocationTimeout = -2`.